handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -479,9 +478,12 @@ public final SSLEngineResult wrap(
                     // It is possible when the outbound was closed there was not enough room in the non-application
                     // buffers to hold the close_notify. We should keep trying to close until we consume all the data
                     // OpenSSL can give us.
-                    boolean didShutdown = doSSLShutdown();
-                    bytesProduced = bioLengthBefore - SSL.bioLengthByteBuffer(networkBIO);
-                    return newResultMayFinishHandshake(didShutdown ? NEED_WRAP : NOT_HANDSHAKING, 0, bytesProduced);
+                    final int shutdownResult = doSSLShutdown();
+                    if (shutdownResult < 0) {
+                        return newResultMayFinishHandshake(NOT_HANDSHAKING, 0, bytesProduced);
+                    }
+                    bytesProduced = bioLengthBefore - shutdownResult;
+                    return newResultMayFinishHandshake(NEED_WRAP, 0, bytesProduced);
```

Comment:
This will not work as `bytesProduced` must be set because it is used to advance the buffer position.